dataline-workers/src/test/java/io/dataline/workers/singer/SingerTapFactoryTest.java

```diff
@@ -0,0 +1,164 @@
+      errorLogPath = jobRoot.resolve(DefaultSyncWorker.TAP_ERR_LOG);
+
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Test
+  public void test() throws InvalidCredentialsException, IOException {
+    StreamFactory streamFactory = mock(StreamFactory.class);
+    SingerMessage recordMessage1 =
+        MessageUtils.createRecordMessage(TABLE_NAME, COLUMN_NAME, "blue");
+    SingerMessage recordMessage2 =
+        MessageUtils.createRecordMessage(TABLE_NAME, COLUMN_NAME, "yellow");
+
+    final List<SingerMessage> expected = Lists.newArrayList(recordMessage1, recordMessage2);
+
+    StandardTapConfig tapConfig =
+        WorkerUtils.syncToTapConfig(TestConfigHelpers.createSyncConfig().getValue());
+    StandardDiscoverSchemaInput discoverSchemaInput = new StandardDiscoverSchemaInput();
+    discoverSchemaInput.setConnectionConfiguration(
+        tapConfig.getSourceConnectionImplementation().getConfiguration());
+    ProcessBuilderFactory pbf = mock(ProcessBuilderFactory.class);
+    ProcessBuilder processBuilder = mock(ProcessBuilder.class);
+    Process process = mock(Process.class);
+    InputStream inputStream = mock(InputStream.class);
+    SingerDiscoverSchemaWorker discoverSchemaWorker = mock(SingerDiscoverSchemaWorker.class);
+
+    final SingerColumn singerColumn = new SingerColumn();
+    singerColumn.setType(Lists.newArrayList(SingerType.NULL, SingerType.STRING));
+
+    final SingerColumnMap singerColumnMap = new SingerColumnMap();
+    singerColumnMap.setAdditionalProperty(COLUMN_NAME, singerColumn);
+
+    final SingerTableSchema singerSchema = new SingerTableSchema();
+    singerSchema.setType("object");
+    singerSchema.setProperties(singerColumnMap);
+
+    SingerStream singerStream = new SingerStream();
+    singerStream.setStream(TABLE_NAME);
+    singerStream.setTapStreamId(TABLE_NAME);
+    singerStream.setTableName(TABLE_NAME);
+    singerStream.setSchema(singerSchema);
+
+    final SingerCatalog singerCatalog = new SingerCatalog();
+    singerCatalog.setStreams(Collections.singletonList(singerStream));
+
+    final OutputAndStatus<SingerCatalog> discoverSchemaOutput =
+        new OutputAndStatus<>(JobStatus.SUCCESSFUL, singerCatalog);
+
+    when(discoverSchemaWorker.runInternal(discoverSchemaInput, discoverSchemaJobRoot))
+        .thenReturn(discoverSchemaOutput);
+
+    when(pbf.create(
+            jobRoot,
+            IMAGE_NAME,
+            "--config",
+            CONFIG_JSON_FILENAME,
+            "--properties",
+            CATALOG_JSON_FILENAME,
+            "--state",
+            STATE_JSON_FILENAME))
+        .thenReturn(processBuilder);
+    when(processBuilder.redirectError(errorLogPath.toFile())).thenReturn(processBuilder);
+    when(processBuilder.start()).thenReturn(process);
+    when(process.getInputStream()).thenReturn(inputStream);
+
+    when(streamFactory.create(any())).thenReturn(expected.stream());
```

Comment:
i don't like the `any()` here but i don't think there's a reasonable equality i can use on `BufferedReader` to get mockito to trigger this on a more tightly specified argument.